src/com/goide/inspections/GoPlaceholderChecker.java

```diff
@@ -0,0 +1,456 @@
+      w = 1;
+      if (placeholderText.charAt(i) == '%') {
+        FormatState state = parsePrintfVerb(placeholderText.substring(i), i, firstArg, argNum);
+        placeholders.add(state.toPlaceholder());
+        w = state.format.length();
+
+        // TODO florin: when is this empty??
+        if (!state.argNums.isEmpty()) {
+          if (!state.indexed) {
+            int maxArgNum = Collections.max(state.argNums) + 1;
+            if (argNum < maxArgNum) {
+              argNum = maxArgNum;
+            }
+          }
+          else {
+            argNum = state.argNums.get(state.argNums.size() - 1) + 1;
+          }
+        }
+      }
+    }
+
+    return placeholders;
+  }
+
+  protected static int getPlaceholderPosition(@NotNull GoFunctionOrMethodDeclaration function) {
+    Integer position = FORMATTING_FUNCTIONS.get(StringUtil.toLowerCase(function.getName()));
+    if (position != null) {
+      String importPath = function.getContainingFile().getImportPath(false);
+      if ("fmt".equals(importPath) || "log".equals(importPath) || TESTING_PATH.equals(importPath)) {
+        return position;
+      }
+    }
+    return -1;
+  }
+
+  @Nullable
+  protected static String resolve(@NotNull GoExpression argument) {
+    if (argument instanceof GoStringLiteral) return argument.getText();
+
+    PsiReference reference = argument.getReference();
+    PsiElement resolved = reference != null ? reference.resolve() : null;
+
+    String value = null;
+    if (resolved instanceof GoVarDefinition) {
+      value = getValue(((GoVarDefinition)resolved).getValue());
+    }
+    else if (resolved instanceof GoConstDefinition) {
+      value = getValue(((GoConstDefinition)resolved).getValue());
+    }
+
+    // We always receive the text with double quotes at the beginning and at the end so remove them
+    if (value != null) {
+      value = value.substring(1, value.length() - 1);
+    }
+
+    return value;
+  }
+
+  // todo: implement ConstEvaluator
+  @Nullable
+  private static String getValue(@Nullable GoExpression expression) {
+    if (expression instanceof GoStringLiteral) {
+      return expression.getText();
+    }
+    if (expression instanceof GoAddExpr) {
+      String sum = getValue((GoAddExpr)expression);
+      return sum.isEmpty() ? null : sum;
+    }
+
+    return null;
+  }
+
+  // todo: implement ConstEvaluator
+  @NotNull
+  private static String getValue(@Nullable GoAddExpr expression) {
+    if (expression == null) return "";
+    StringBuilder result = new StringBuilder();
+    for (GoExpression expr : expression.getExpressionList()) {
+      if (expr instanceof GoStringLiteral) {
+        result.append(expr.getText());
+      }
+      else if (expr instanceof GoAddExpr) {
+        result.append(getValue(expr));
+      }
+    }
+
+    return result.toString();
+  }
+
+  private static class FormatState {
+    private char verb;                  // the format verb: 'd' for "%d"
```

Comment:
As I can see this field is not used much. It's compared to `%` and it's converted to Verb. So why it's not nullable Verb field?